crates/anvil/src/config.rs

```diff
@@ -834,7 +834,13 @@ impl NodeConfig {
 
         // if provided use all settings of `genesis.json`
         if let Some(ref genesis) = self.genesis {
-            genesis.apply(&mut env);
+            env.cfg.chain_id = genesis.config.chain_id;
+            env.block.timestamp = U256::from(genesis.timestamp);
+            if let Some(base_fee) = genesis.base_fee_per_gas {
+                env.block.basefee = U256::from(base_fee);
+            }
+            env.block.number = U256::from(genesis.nonce);
```

Comment:
this is not quite right

I think we're also missing optional number on alloy genesis:

https://github.com/ethereum/go-ethereum/blob/fc380f52ef9778e988266f776b9593ce719cf79d/core/genesis.go#L64-L64

Reply:
yep, let's add it.

Reply:
https://github.com/foundry-rs/foundry/pull/6970/commits/f0df51e06598db282cd630b8ab050015fdfb06de